File: 7KYU/A Rule of Divisibility by 7/solution.clj
(ns divpar7.core)

(defn seven [m]
  (loop [n m
         steps 0]
    (if (< (count (str n)) 3)
      [n steps]
      (recur (- (quot n 10) (*' 2 (mod n 10))) (inc steps)))))

File: 7KYU/A Rule of Divisibility by 7/solution.go
package kata

import "fmt"

func Seven(n int64) []int {
	steps := 0
	for {
		sn := len(fmt.Sprint(n))
		if sn < 3 {
			return []int{int(n), steps}
		}

		n = (n / 10) - (2 * (n % 10))
		steps += 1
	}
}
